agents/go-agents/core/process/process_cleaner.go

```diff
@@ -42,21 +42,32 @@ func (pc *Cleaner) CleanPeriodically() {
 	}
 }
 
-// CleanOnce cleanups processes that exited more than CleanupThreshold time ago
+// CleanOnce cleanups processes that died before Time.Now() minus CleanupThreshold.
+//
+//  process1.deathTime = 2
+//  process2.deathTime = 5
+//
+//                    death bound           now
+//                         v                 v
+// timeline -> 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 ->
```

Comment:
This scheme makes it really clear - good idea! 
BTW can you consider the addition of CleanupThreshold value that is used in pseudographics?